Automattic/wp-calypso
# Gutenboarding: Display FSE design using seedlet-blocks

This resolves #43288, since you will be able to activate seedlet-blocks internally via Gutenboarding in staging environments. This also resolves #43285, which adds a design which when chosen, will activate FSE on the new site.

This diff is the companion on wpcom and has already been deployed: D45075-code.

### Changes proposed in this Pull Request
* Add `is_fse` flag and `seedlet-blocks` theme to a design
* Fix issue where selecting a premium plan did not redirect you to the site editor
* Added selector for FSE site now that there are multiple ways to enter the flow
* Instead of signup flow option, use new `enabled_fse` option to create the site. This way, the backend still considers the ultimate flow to be gutenboarding. (see code-D45075 as well.)
* When the `gutenboarding/site-editor` flag is set, convert every design to use `is_fse` and `seedlet-blocks` as the theme.

~This is currently blocked on #43415, which needs deployed so that the seedlet demo site begins working again. Once that is working, we can correctly generate previews. Currently previews do not work because the demo site does not render a post content block:~ (Previews are now working.)

~Second, we need the `is_edge` config option in #42354 to restrict the design to certain environents. Otherwise this will start working in prod.~ (This has been completed.)

### Known issues:
* Font options do not work with the `seedlet-blocks` theme.
* When loading the site editor for the first time, the incorrect template is used for the homepage, meaning that you do not see the post content of your selected design. This is fixed by https://github.com/WordPress/gutenberg/pull/23186, which will arrive on dotcom in the coming weeks.

### Testing instructions
* (if you are sandboxed, make sure it is up to date. being sandboxed is no longer a requirement)
* Go through `http://calypso.localhost:3000/new`
* Select the first design option (named "FSE Test Design")
* Select the premium plan
* Verify that you are redirected to the site editor after getting the premium plan
* Verify that the seedlet theme is active on your site.
* Verify that the `gutenberg-edge` sticker is applied to your site.
* Verify that the homepage post_content matches the Cassel page pattern (it should _not_ match the seedlet-blocks default homepage content.)

Ålso test:
* Use http://calypso.localhost:3000/new?flags=gutenboarding/site-editor
* Every design option should have the seedlet-blocks theme set and the previews should look correct.
* Finishing the flow should land in the site editor, with the selected design's page pattern showing on the front end.
